lib/reporters/tap.js

```diff
@@ -75,24 +88,25 @@ TapReporter.prototype.unhandledError = function (err) {
 		);
 	}
 
-	return output.join('\n');
+	this.write(output.join('\n'));
 };
 
-TapReporter.prototype.finish = function (runStatus) {
+TapReporter.prototype.finish = function () {
+	var status = this.status;
 	var output = [
 		'',
-		'1..' + (runStatus.passCount + runStatus.failCount + runStatus.skipCount),
-		'# tests ' + (runStatus.passCount + runStatus.failCount + runStatus.skipCount),
-		'# pass ' + runStatus.passCount
+		'1..' + (status.passCount + status.failCount + status.skipCount),
+		'# tests ' + (status.passCount + status.failCount + status.skipCount),
+		'# pass ' + status.passCount
 	];
 
-	if (runStatus.skipCount > 0) {
-		output.push('# skip ' + runStatus.skipCount);
+	if (status.skipCount > 0) {
+		output.push('# skip ' + status.skipCount);
 	}
 
-	output.push('# fail ' + (runStatus.failCount + runStatus.rejectionCount + runStatus.exceptionCount), '');
+	output.push('# fail ' + (status.failCount + status.rejectionCount + status.exceptionCount), '');
 
-	return output.join('\n');
+	this.write(output.join('\n'));
 };
 
 TapReporter.prototype.write = function (str) {
```

Comment:
why do we need TapReporter.prototype.stdout and TapReporter.prototype.stderr if they're never used here?